### PR TITLE
giflib 4.2.3

### PR DESCRIPTION
**Pull request by nijikon**
This PR update giflib to 4.2.3 and all brews that uses it. Feel free to comment if I done everything correctly.

### CODE REVIEW

**Comment by nijikon** [comment]
Fixes #40587

File: Library/Formula/mapserver.rb [hunk 1]
```diff
@@ -31,6 +30,8 @@ class Mapserver < Formula
   depends_on 'fcgi' if build.with? "fastcgi"
   depends_on 'cairo' => :optional
 
+  patch :DATA
```
Comment (DomT4) [inline Library/Formula/mapserver.rb]: Can you add a comment explaining what this patch does and where it's been reported or committed upstream? Helps us know when it's safe to get rid of. Thanks!

File: Library/Formula/xplanet.rb [hunk 2]
```diff
@@ -41,6 +39,18 @@ class Xplanet < Formula
   depends_on "freetype"
   depends_on :x11 => :optional
 
+  # support giflib 4.2.x
```
Comment (DomT4) [inline Library/Formula/xplanet.rb]: Likewise these need submitting upstream ideally.

Comment (nijikon) [inline Library/Formula/xplanet.rb]: It is from upstream, no new version of xplanet that is why I used this patch.

Comment (DomT4) [inline Library/Formula/xplanet.rb]: Ah cool. Can you add a link to the upstream commit or something?

File: Library/Formula/mapserver.rb (hunk 1 above)
Comment (nijikon) [inline Library/Formula/mapserver.rb]: Done.

File: Library/Formula/xplanet.rb (hunk 2 above)
Comment (nijikon) [inline Library/Formula/xplanet.rb]: Done

**Comment by nijikon** [comment]
@DomT4 can you reschedule this?

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Still getting this weird test failure:

```
Failed to read test report file /Users/brew/Jenkins/prs/version/yosemite/brew-test-bot.xml
org.dom4j.DocumentException: Error on line 22092 of document file:///Users/brew/Jenkins/prs/version/yosemite/brew-test-bot.xml : An invalid XML character (Unicode: 0x15) was found in the CDATA section. Nested exception: An invalid XML character (Unicode: 0x15) was found in the CDATA section.
```

The build is fine. The only failure on building is FontForge flagging that it links into Python Frameworks, which is a known issue we're working on and can be ignored, but I don't want to pull this until we know the `brew-test-bot.xml.<init>` failure is safe to ignore.

CC @mikemcquaid @jacknagel

**Comment by nijikon** [comment]
@jacknagel @mikemcquaid can we do something with this? It prevents me from upgrading gdal to 2.0 version. Thanks!

**Comment by mikemcquaid** [comment]
@nijikon Thanks, pushed! Check out the commit message edits I made for next time (basically: always use the format `formula: blah blah blah`)

**Comment by sanak** [comment]
I encountered the following error.
<pre>
$ psql -U postgres -d test -c "CREATE EXTENSION postgis;"
ERROR:  could not load library "/usr/local/Cellar/postgresql/9.4.4/lib/rtpostgis-2.1.so": dlopen(/usr/local/Cellar/postgresql/9.4.4/lib/rtpostgis-2.1.so, 10): Symbol not found: _PrintGifError
  Referenced from: /usr/local/lib/libgdal.1.dylib
  Expected in: /usr/local/lib/libgif.4.dylib
 in /usr/local/lib/libgdal.1.dylib
</pre>
I think that gdal bottle should be updated to use new giflib version.
Should I create a new ticket for gdal ?

**Comment by nijikon** [comment]
@sanak Thanks! I'm currently working on gdal update.

**Comment by DomT4** [comment]
Thanks for chasing up on that @nijikon, and for reporting the issue @sanak. Appreciate the quick community response :heart:

**Comment by sanak** [comment]
@nijikon @DomT4 Okay, thanks for confirmation!

**Pull request closed without merging**